e2e/testcases/admission_test.go

```diff
@@ -141,55 +147,96 @@ metadata:
-	}
-
-	// Recreate the admission webhook
-	if _, err := nt.Shell.Kubectl("replace", "-f", filepath.Join(nt.TmpDir, "webhook.yaml")); err != nil {
-		nt.T.Fatalf("failed to replace the admission webhook %v", err)
-	}
-
-	// Verify that the webhook is disabled.
-	if _, err := nt.Shell.Kubectl("delete", "ns", "hello"); err != nil {
-		nt.T.Fatalf("failed to run `kubectl delete ns hello` %v", err)
-	}
-
-	// Remove the annotation for disabling webhook
-	if _, err := nt.Shell.Kubectl("annotate", "ValidatingWebhookConfiguration", "admission-webhook.configsync.gke.io", "configsync.gke.io/webhook-configuration-update-"); err != nil {
-		nt.T.Fatalf("failed to remove the annotation in the admission webhook %v", err)
+	//helloNS := &corev1.Namespace{}
+	//if err := nt.KubeClient.Get("hello", "", helloNS); err != nil {
+	//	nt.T.Fatal(err)
+	//}
+	//version := helloNS.ResourceVersion
+
+	nomostest.StopWebhook(nt)
+
+	tg := taskgroup.New()
+	tg.Go(func() error {
+		predicates := []testpredicates.Predicate{
+			testpredicates.StatusEquals(nt.Scheme, kstatus.CurrentStatus),
+			testpredicates.DeploymentMissingEnvVar(reconcilermanager.Reconciler, reconcilermanager.WebhookEnabled),
+		}
+		return nt.Watcher.WatchObject(kinds.Deployment(),
+			core.RootReconcilerName(configsync.RootSyncName), configsync.ControllerNamespace, predicates)
+	})
+	tg.Go(func() error {
+		return nt.Watcher.WatchObject(kinds.Namespace(), "hello", "",
+			[]testpredicates.Predicate{
+				testpredicates.MissingAnnotation(metadata.DeclaredFieldsKey),
+			})
+	})
+	if err := tg.Wait(); err != nil {
+		nt.T.Fatal(err)
 	}
-
-	nt.Must(nt.RootRepos[configsync.RootSyncName].Add("acme/namespaces/test/ns.yaml", fake.NamespaceObject("test")))
-	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("add another Namespace"))
-	if err := nt.WatchForAllSyncs(); err != nil {
+	//nt.T.Logf("Check if the namespace resourceversion has changed after deleting Admission Webhook")
+	//err := nt.Watcher.WatchObject(kinds.Namespace(), "hello", "",
+	//	[]testpredicates.Predicate{
+	//		testpredicates.ResourceVersionNotEquals(nt.Scheme, version),
+	//	})
+	//if err != nil {
+	//	nt.T.Fatal(err)
+	//}
+	//nt.T.Logf("Check if the test namespace has dropped the declared-field annotation")
+	//if err := nt.Validate("hello", "", &corev1.Namespace{}, testpredicates.MissingAnnotation(metadata.DeclaredFieldsKey)); err != nil {
+	//	nt.T.Fatal(err)
+	//}
+
+	//helloNS = &corev1.Namespace{}
+	//if err := nt.KubeClient.Get("hello", "", helloNS); err != nil {
+	//	nt.T.Fatal(err)
+	//}
+	//version = helloNS.ResourceVersion
+
+	if err := nomostest.InstallWebhook(nt); err != nil {
 		nt.T.Fatal(err)
 	}
-
 	nomostest.WaitForWebhookReadiness(nt)
 
-	// Verify that the webhook is now enabled
-	if _, err := nt.Shell.Kubectl("delete", "ns", "test"); err == nil {
-		nt.T.Fatal("got `kubectl delete ns hello` success, want return err")
+	nt.T.Logf("Check declared-fields annotation is missing")
```

Comment:
"Check declared-fields annotation is missing" -> "Check declared-fields annotation is re-populated"